packages/core/lib/package.js

```diff
@@ -1,351 +1,214 @@
+      storage: "ethpm/storage/ipfs"
+    }).connect({
+      provider: provider,
+      workingDirectory: options.working_directory,
+      registryAddress: ethpmURI.address,
+      ipfs: {
+        host: options.ethpm.ipfs_host,
+        port: options.ethpm.ipfs_port,
+        protocol: options.ethpm.ipfs_protocol
+      }
+    });
 
-    // When installing, we use infura to make a bunch of eth_call's.
-    // We don't make any transactions. To satisfy APIs we'll put a from address,
-    // but it doesn't really matter in this case.
-    const fakeAddress = "0x1234567890123456789012345678901234567890";
-
-    let registry = options.ethpm.registry;
-
-    if (typeof registry === "string") {
-      registry = await EthPMRegistry.use(
-        options.ethpm.registry,
-        fakeAddress,
-        provider
+    // Validate target package is available on connected registry
+    const availablePackages = await ethpm.registries.packages();
+    if (!availablePackages.includes(ethpmURI.packageName)) {
+      throw new TruffleError(
+        "Package: " +
+          ethpmURI.packageName +
+          ", not found on registry at address: " +
+          ethpmURI.address +
+          ". Available packages include: " +
+          availablePackages
       );
     }
 
-    const pkg = new EthPM(options.working_directory, host, registry);
-
-    if (options.packages) {
-      const promises = options.packages.map(package_name => {
-        const pieces = package_name.split("@");
-        package_name = pieces[0];
-
-        let version = "*";
-
-        if (pieces.length > 1) version = pieces[1];
-
-        return pkg.installDependency(package_name, version);
-      });
-
-      Promise.all(promises)
-        .then(() => {
-          if (options.packages.length > 0) {
-            console.log("");
-            console.log("Successfully installed the following package(s)...");
-            console.log("==================================================");
-            options.packages.forEach(singlePackage => {
-              console.log(`> ${singlePackage}`);
-            });
-            console.log("");
-          }
-          callback();
-        })
-        .catch(callback);
-    } else {
-      fs.access(
-        path.join(options.working_directory, "ethpm.json"),
-        fs.constants.R_OK,
-        error => {
-          let manifest;
-
-          // If the ethpm.json file doesn't exist, use the config as the manifest.
-          if (error) manifest = options;
-
-          pkg
-            .install(manifest)
-            .then(() => {
-              callback();
-            })
-            .catch(callback);
-        }
-      );
-    }
+    // Fetch target package manifest URI
+    const manifestURI = await ethpm.registries
+      .package(ethpmURI.packageName)
+      .release(ethpmURI.version);
+
+    // Install target manifest URI to working directory
+    await ethpm.installer.install(manifestURI, ethpmURI.address);
```

Comment:
we need to capture the result here as a list of contract types and list of deployments, so that we can pass this list of contracts to @truffle/artifactor. 

you'll need convert blockchain URIs into network IDs because that's how the artifacts format organizes contract instances

Reply:
note that the format is formally spec'd in @truffle/contract-schema